tests/test_zero_data_parallel/test_zero_engine.py

```diff
@@ -30,10 +33,16 @@ def run_dist(rank, world_size, port, parallel_config):
     for model_name in test_models:
         get_components_func = non_distributed_component_funcs.get_callable(model_name)
         model_builder, train_dataloader, _, optimizer_class, criterion = get_components_func()
-
-        colo_model = model_builder(checkpoint=True)
-        torch_model = copy.deepcopy(colo_model).cuda()
-        torch_model.train()
+        with ZeroInitContext(convert_fp16=hasattr(gpc.config, 'fp16'),
+                             target_device=torch.cuda.current_device(),
+                             shard_strategy=gpc.config.zero.model_config.shared_strategy(
+                                 gpc.get_group(ParallelMode.DATA)),
+                             shard_param=True):
+            colo_model = model_builder(checkpoint=True)
+
+        torch_model = model_builder(checkpoint=True).half()
+        col_model_deepcopy(colo_model, torch_model)
+        torch_model = torch_model.cuda().float()
```

Comment:
why .float()?

Reply:
Because we don't convert float inputs to fp16. We will refactor this part later. Maybe we should compare zero with apex O1 amp.

Reply:
TYPO: We should use apex O2 amp.

Reply:
compare with O1 in the next MR